src/oatpp/core/base/Environment.cpp

```diff
@@ -270,15 +294,34 @@ void Environment::log(v_int32 priority, const std::string& tag, const std::strin
 }
 
 void Environment::logFormatted(v_int32 priority, const std::string& tag, const char* message, ...) {
+  // do we have a logger and the priority is enabled?
+  if (m_logger == nullptr || !m_logger->isLogPriorityEnabled(priority)) {
+    return;
+  }
+  // if we dont need to format anything, just print the message
   if(message == nullptr) {
-    message = "[null]";
+    log(priority, tag, std::string());
+    return;
   }
-  char buffer[4097];
+  // check how big our buffer has to be
   va_list args;
-  va_start (args, message);
-  vsnprintf(buffer, 4096, message, args);
+  va_start(args, message);
+  v_buff_size allocsize = vsnprintf(nullptr, 0, message, args) + 1;
+  va_end(args);
+  // alloc the buffer (or the max size)
+  if (allocsize > m_logger->getMaxFormattingBufferSize()) {
+    allocsize = m_logger->getMaxFormattingBufferSize();
+  }
+  char* buffer = new char[allocsize];
```

Comment:
Let's use the `unique_ptr` here, as the user-provided `log()` may potentially throw.